### PR TITLE
Add executable plugins

### PR DESCRIPTION
**Pull request by roose**
Add executable plugins, for @happlebao, please look at this :smile: Just for example )

### CODE REVIEW

**Comment by bao-qian** [comment]
The python plugin works because there is a python wrapper called wox.py located in jsonrpc.  Then how could you write a wrapper for executable plugin?

**Comment by roose** [comment]
Just with send json, `{"result":[items]}` etc, developers let make own wrappers, as example I began write a wrapper for Nimlang https://github.com/roose/nim-wox

File: Wox.Core/Plugin/PluginsLoader.cs [hunk 1]
```diff
@@ -110,5 +110,18 @@ public static IEnumerable<PluginPair> PythonPlugins(IEnumerable<PluginMetadata>
...
             return plugins;
         }
+
+        public static IEnumerable<PluginPair> ExecutablePlugins(IEnumerable<PluginMetadata> source)
+        {
+            var metadatas = source.Where(o => o.Language.ToUpper() == AllowedLanguage.Executable);
+
+            var plugins = metadatas.Select(metadata => new PluginPair
+            {
+                Plugin = new ExecutablePlugin(metadata.ExecuteFilePath),
```
Comment (bao-qian) [inline Wox.Core/Plugin/PluginsLoader.cs:120]: The arguments for constructor is the executable for that language (pythonw etc) , not the executable for plugin (main.lan)

**Comment by bao-qian** [comment]
As I said above, if you know the executable name(nim), then why not call it NimPlugin?

**Comment by roose** [comment]
It's just for example, i don't know C#.
Nim is language, not plugin, I want write exe plugin, e.g. `main.exe` and call it from wox:
woxnim
e.g. I just send json to `main.exe` and get results with jsonrpc

**Comment by bao-qian** [comment]
Ah, get it, nim is not a interpretative language, it needs to compile it self right?

**Comment by roose** [comment]
yes ) sorry for my english =)

**Comment by bao-qian** [comment]
Don't worry, I'm not a native speaker too.
That's make sense. 
Remember to add wox.nim to JsonRPC directory in the future.

But I must warn you the I will change the API structure in the near future, so you will need to rewrite it.

**Comment by roose** [comment]
No, Wox don't need wox.nim it's just for nim developers, e.g. in nim program developer can import it, e.g. `import wox.nim` it's wrapper for make jsonrpc return when wox call exe plugin, just as python library for importing it

**Comment by bao-qian** [comment]
I know, but how could other nim developers know that the nim wrapper is existed? I put wox.py  there for the same reason.

**Comment by bao-qian** [comment]
**All checks have failed**
fix this, then I will merge it.

**Comment by roose** [comment]
nimlang have own package manager, after I finish it and publish, nim developer just run `nimble install wox` and library downloaded and placed in special dir, in next time developer just type `import wox`

**Comment by bao-qian** [comment]
I wants to centralize all the things.
since you probably don't want pull the whole wox to maintain the nim language,

I'd suggest I add you as a collarborator in this repo: https://github.com/Wox-launcher/LanguageBinding

then add wox.nim there

I'd appreciate if you could also public the python binding.

**Comment by roose** [comment]
Ok, thanks for suggestion :smile:, as soon as I finish wrapper I publish it.
About python binding, you're talking about current binding?

Hmm, I do not understand, how I can see what the failed? =)

**Comment by bao-qian** [comment]
<del >https://github.com/Wox-launcher/Wox/pull/597#commitcomment-17389408 </del>

you don't have vs in your machine, right..?

**Comment by roose** [comment]
I have VS Express, but I don't have any errors when compiled this code

**Comment by bao-qian** [comment]
fine, I will cherry-pick this pr.

> About python binding, you're talking about current binding?

yeah

> Ok, thanks for suggestion :smile:, as soon as I finish wrapper I publish it.

Since I added you as collaborator, I wants you **develop** your wrapper there, and same for auto publish, continue integration etc.

**Comment by bao-qian** [comment]
not sure why, could you give it a try by open a new pr?

**Comment by roose** [comment]
Check failed may be because I make branch from master and send pr to dev?

I need delete https://github.com/roose/nim-wox and work only with https://github.com/Wox-launcher/LanguageBinding ? Even if the wrapper don't finished? Or I work with own repo and send changes only stable releases?

Ok, I'm going to try now

**Comment by bao-qian** [comment]
> I need delete https://github.com/roose/nim-wox and work only with https://github.com/Wox-launcher/LanguageBinding ? Even if the wrapper don't finished?

yes

File: Wox.Core/Plugin/ExecutablePlugin.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            };
+
+            _startInfo.Arguments = $"\"{request}\"";
+
+            return Execute(_startInfo);
+        }
+
+        protected override string ExecuteCallback(JsonRPCRequestModel rpcRequest)
+        {
+            _startInfo.Arguments = $"\"{context.CurrentPluginMetadata.ExecuteFilePath}\" \"{rpcRequest}\"";
```
Comment (bao-qian) [inline Wox.Core/Plugin/ExecutablePlugin.cs:42]: since your are using .exe directly, you don't need to add it to arguments

**Comment by roose** [comment]
yes, I forgot change this, I now testing on dev branch, if all ok I send pr

**Comment by roose** [comment]
@happlebao oops, dev branch don't compile, bunch of errors, what the branch I need to change? Or just send PR without compiling?

**Comment by bao-qian** [comment]
try again

**Pull request closed without merging**